fvm/errors/errors.go

```diff
@@ -180,6 +257,9 @@ func WrapCodedError(
 	prefixMsgFormat string,
 	formatArguments ...interface{},
 ) codedError {
+	if code.IsFailure() {
+		panic(fmt.Sprintf("cannot wrap failure (%s) as CodedError", code))
+	}
```

Comment:
This is not something that is visible compile time and might be a sneaky time bomb once deployed on mainnet.

I would prefer `ErrorCode`s being separated into ErrorCode and `FailureCode` so that newError can only accept ErrorCodes, making it much harder to make a mistake that could result in the node crashing unexpectedly in production.